pkg/cloudprovider/providers/gce/gce_loadbalancer_external_test.go

```diff
@@ -953,3 +953,98 @@ func TestEnsureExternalLoadBalancerDeletedSucceedsOnXPN(t *testing.T) {
 	msg := fmt.Sprintf("%s %s %s", v1.EventTypeNormal, eventReasonManualChange, eventMsgFirewallChange)
 	checkEvent(t, recorder, msg, true)
 }
+
+type EnsureELBParams struct {
+	clusterName, clusterID string
+	apiService             *v1.Service
+	existingFwdRule        *compute.ForwardingRule
+	nodes                  []*v1.Node
+}
+
+// newEnsureELBParams is the constructor of EnsureELBParams.
+func newEnsureELBParams(nodes []*v1.Node) *EnsureELBParams {
+	vals := DefaultTestClusterValues()
+	return &EnsureELBParams{
+		vals.ClusterName,
+		vals.ClusterID,
+		fakeApiService.DeepCopy(),
+		nil,
+		nodes,
+	}
+}
+
+// callEnsureExternalLoadBalancerAndCheckError wraps calling for
+// ensureExternalLoadBalancer, so TestEnsureExternalLoadBalancerErrors can be
+// easily maintained if the interface changes.
+func callEnsureExternalLoadBalancerAndCheckError(t *testing.T, gce *GCECloud, params *EnsureELBParams) {
+	status, err := gce.ensureExternalLoadBalancer(
+		params.clusterName,
+		params.clusterID,
+		params.apiService,
+		params.existingFwdRule,
+		params.nodes,
+	)
+	assert.Error(t, err)
+	assert.Nil(t, status)
+}
+
+func TestEnsureExternalLoadBalancerErrors(t *testing.T) {
```

Comment:
Maybe add a comment explaining this is testing that "it doesn't panic when errors raised"? That is not obvious.

Reply:
Comment added. Vague function name indeed. Let me know if you have a better name for the function in you mind.